modules/gateway/gateway.go

```diff
@@ -159,18 +145,26 @@ func New(addr string, persistDir string) (g *Gateway, err error) {
 	g.log.Println("INFO: gateway created, started logging")
 
 	// Forward the RPC port, if possible.
-	go g.forwardPort(port)
+	g.threads.Add()
+	go func() {
+		defer g.threads.Done()
+		g.forwardPort(port)
+	}()
 
 	// Learn our external IP.
-	go g.learnHostname(port)
+	g.threads.Add()
+	go func() {
+		defer g.threads.Done()
```

Comment:
Always check your error values, doesn't matter if you know it can't return an error. Knowing that something can't return an error is an implementation detail,which means your code is brittle against implementation changes.

We've had enough bugs crop up from unchecked errors that I feel we need to be pedantic about it.